swc_ecma_transforms_base v0.89.5 - fails to build

### Describe the bug

Not sure what's going on, but swc_ecma_transforms_base 0.89.5 is failing to build https://github.com/denoland/deno_ast/pull/100 -- maybe it's related to me moving away from swc_ecmascript. I'm not sure, but feel like this should still be able to build.

Rust: 1.61.0

```
Run cargo clippy --all-targets --all-features --release
  cargo clippy --all-targets --all-features --release
  deno lint
  shell: /usr/bin/bash -e {0}
  env:
    CARGO_INCREMENTAL: 0
    GH_ACTIONS: 1
    RUST_BACKTRACE: full
    RUSTFLAGS: -D warnings
    CACHE_ON_FAILURE: false
    Updating crates.io index
 Downloading crates ...
  Downloaded cfg-if v1.0.0
  Downloaded itertools v0.10.3
  Downloaded phf_shared v0.10.0
  Downloaded proc-macro-error v1.0.4
  Downloaded rand_core v0.6.3
  Downloaded proc-macro2 v1.0.40
  Downloaded precomputed-hash v0.1.1
  Downloaded pmutil v0.5.3
  Downloaded scoped-tls v1.0.0
  Downloaded scopeguard v1.1.0
  Downloaded semver-parser v0.7.0
  Downloaded serde_derive v1.0.137
  Downloaded sha-1 v0.10.0
  Downloaded relative-path v1.7.0
  Downloaded string_cache_codegen v0.5.2
  Downloaded aho-corasick v0.7.18
  Downloaded parking_lot v0.12.1
  Downloaded swc_ecma_visit v0.65.0
  Downloaded swc_ecma_parser v0.105.6
  Downloaded swc_trace_macro v0.1.1
  Downloaded swc_graph_analyzer v0.7.0
  Downloaded text_lines v0.4.1
  Downloaded num_cpus v1.13.1
  Downloaded static-map-macro v0.2.3
  Downloaded strsim v0.9.3
  Downloaded rayon-core v1.9.3
  Downloaded petgraph v0.6.2
  Downloaded crc v2.1.0
  Downloaded darling_macro v0.10.2
  Downloaded crc-catalog v1.1.1
  Downloaded if_chain v1.0.2
  Downloaded ident_case v1.0.1
  Downloaded matches v0.1.9
  Downloaded auto_impl v0.5.0
  Downloaded darling v0.10.2
  Downloaded lexical-core v0.8.5
  Downloaded crossbeam-channel v0.5.5
  Downloaded swc_macros_common v0.3.5
  Downloaded swc_ecma_transforms_compat v0.104.1
  Downloaded lexical-util v0.8.5
  Downloaded swc_ecma_transforms v0.161.0
  Downloaded swc_ecma_transforms_typescript v0.124.0
  Downloaded lexical-write-integer v0.8.5
  Downloaded radix_fmt v1.0.0
  Downloaded libc v0.2.126
  Downloaded swc_ecma_transforms_base v0.89.5
  Downloaded swc_bundler v0.155.0
  Downloaded dashmap v5.3.4
  Downloaded st-map v0.1.6
  Downloaded string_cache v0.8.4
  Downloaded crossbeam-utils v0.8.9
  Downloaded swc_ecma_transforms_react v0.120.0
  Downloaded percent-encoding v2.1.0
  Downloaded rustc_version v0.2.3
  Downloaded swc_ecma_ast v0.79.0
  Downloaded swc_config_macro v0.1.0
  Downloaded rand_chacha v0.3.1
  Downloaded thiserror v1.0.31
  Downloaded swc_ecma_loader v0.30.2
  Downloaded swc_ecma_codegen v0.109.1
  Downloaded tracing-core v0.1.27
  Downloaded unicode-width v0.1.9
  Downloaded swc_ecma_transforms_classes v0.77.0
  Downloaded unreachable v0.1.1
  Downloaded url v2.2.2
  Downloaded tracing-attributes v0.1.21
  Downloaded swc_eq_ignore_macros v0.1.0
  Downloaded void v1.0.2
  Downloaded unicode-id v0.3.2
  Downloaded tracing v0.1.35
  Downloaded version_check v0.9.4
  Downloaded swc_visit v0.3.0
  Downloaded thiserror-impl v1.0.31
  Downloaded swc_visit_macros v0.3.1
  Downloaded swc_ecma_transforms_proposal v0.112.0
  Downloaded swc_ecma_transforms_macros v0.5.0
  Downloaded swc_ecma_dep_graph v0.77.0
  Downloaded typenum v1.15.0
  Downloaded tinyvec_macros v0.1.0
  Downloaded time v0.1.44
  Downloaded typed-arena v2.0.1
  Downloaded crossbeam-epoch v0.9.9
  Downloaded unicode-normalization v0.1.19
  Downloaded unicode-ident v1.0.1
  Downloaded unicode-bidi v0.3.8
  Downloaded tinyvec v1.6.0
  Downloaded sourcemap v6.0.2
  Downloaded syn v1.0.98
  Downloaded semver v0.9.0
  Downloaded ryu v1.0.10
  Downloaded proc-macro-error-attr v1.0.4
  Downloaded once_cell v1.12.0
  Downloaded swc_common v0.18.9
  Downloaded swc_cached v0.1.1
  Downloaded swc_atoms v0.2.12
  Downloaded string_enum v0.3.1
  Downloaded serde v1.0.137
  Downloaded rand v0.8.5
  Downloaded pretty_assertions v1.2.1
  Downloaded pin-project-lite v0.2.9
  Downloaded parking_lot_core v0.9.3
  Downloaded num-traits v0.2.15
  Downloaded lexical v6.1.1
  Downloaded base64 v0.11.0
  Downloaded quote v1.0.20
  Downloaded chrono v0.4.19
  Downloaded bumpalo v3.10.0
  Downloaded static_assertions v1.1.0
  Downloaded num-integer v0.1.45
  Downloaded fnv v1.0.7
  Downloaded swc_ecma_minifier v0.122.0
  Downloaded rayon v1.5.3
  Downloaded browserslist-rs v0.11.0
  Downloaded lexical-write-float v0.8.5
  Downloaded better_scoped_tls v0.1.0
  Downloaded swc_ecma_transforms_optimization v0.131.0
  Downloaded swc_ecma_preset_env v0.136.0
  Downloaded ppv-lite86 v0.2.16
  Downloaded ordered-float v2.10.0
  Downloaded lexical-parse-float v0.8.5
  Downloaded ast_node v0.7.7
  Downloaded data-url v0.1.1
  Downloaded darling_core v0.10.2
  Downloaded crossbeam-deque v0.8.1
  Downloaded base64 v0.13.0
  Downloaded ahash v0.7.6
  Downloaded Inflector v0.11.4
  Downloaded debug_unreachable v0.1.1
  Downloaded crypto-common v0.1.3
  Downloaded num-bigint v0.4.3
  Downloaded minimal-lexical v0.2.1
  Downloaded retain_mut v0.1.9
  Downloaded proc-macro-hack v0.5.19
  Downloaded pathdiff v0.2.1
  Downloaded new_debug_unreachable v1.0.4
  Downloaded from_variant v0.1.3
  Downloaded dprint-swc-ext v0.2.0
  Downloaded lexical-parse-integer v0.8.6
  Downloaded is-macro v0.2.1
  Downloaded enum_kind v0.2.1
  Downloaded swc_fast_graph v0.6.0
  Downloaded regex-syntax v0.6.26
  Downloaded swc_timer v0.6.0
  Downloaded swc_ecma_utils v0.86.2
  Downloaded swc_ecma_codegen_macros v0.7.1
  Downloaded swc_config v0.1.1
  Downloaded memoffset v0.6.5
  Downloaded memchr v2.5.0
  Downloaded lazy_static v1.4.0
  Downloaded itoa v1.0.2
  Downloaded indexmap v1.9.0
  Downloaded idna v0.2.3
  Downloaded hashbrown v0.12.1
  Downloaded getrandom v0.2.7
  Downloaded generic-array v0.[14](https://github.com/denoland/deno_ast/runs/6976588917?check_suite_focus=true#step:7:14).5
  Downloaded form_urlencoded v1.0.1
  Downloaded fixedbitset v0.4.1
  Downloaded either v1.6.1
  Downloaded digest v0.10.3
  Downloaded diff v0.1.12
  Downloaded cpufeatures v0.2.2
  Downloaded block-buffer v0.10.2
  Downloaded bitflags v1.3.2
  Downloaded autocfg v1.1.0
  Downloaded anyhow v1.0.58
  Downloaded ansi_term v0.12.1
  Downloaded smallvec v1.8.0
  Downloaded siphasher v0.3.10
  Downloaded rustc-hash v1.1.0
  Downloaded nom v7.1.1
  Downloaded arrayvec v0.7.2
  Downloaded preset_env_base v0.2.4
  Downloaded serde_json v1.0.81
  Downloaded semver v1.0.10
  Downloaded lock_api v0.4.7
  Downloaded regex v1.5.6
  Downloaded phf_macros v0.10.0
  Downloaded phf_generator v0.10.0
  Downloaded phf v0.10.1
   Compiling proc-macro2 v1.0.40
   Compiling quote v1.0.20
   Compiling unicode-ident v1.0.1
   Compiling syn v1.0.98
   Compiling autocfg v1.1.0
   Compiling serde_derive v1.0.137
   Compiling version_check v0.9.4
   Compiling serde v1.0.137
    Checking cfg-if v1.0.0
   Compiling memchr v2.5.0
   Compiling libc v0.2.126
    Checking once_cell v1.12.0
    Checking scopeguard v1.1.0
   Compiling ppv-lite86 v0.2.16
    Checking hashbrown v0.12.1
   Compiling parking_lot_core v0.9.3
   Compiling siphasher v0.3.10
    Checking smallvec v1.8.0
   Compiling serde_json v1.0.81
   Compiling regex-syntax v0.6.26
    Checking ryu v1.0.10
    Checking tinyvec_macros v0.1.0
    Checking new_debug_unreachable v1.0.4
    Checking itoa v1.0.2
    Checking matches v0.1.9
    Checking either v1.6.1
    Checking precomputed-hash v0.1.1
   Compiling ident_case v1.0.1
   Compiling lazy_static v1.4.0
   Compiling strsim v0.9.3
   Compiling fnv v1.0.7
    Checking unicode-bidi v0.3.8
    Checking pin-project-lite v0.2.9
    Checking percent-encoding v2.1.0
    Checking rustc-hash v1.1.0
    Checking void v1.0.2
    Checking base64 v0.11.0
    Checking if_chain v1.0.2
    Checking scoped-tls v1.0.0
    Checking unicode-width v0.1.9
    Checking static_assertions v1.1.0
   Compiling anyhow v1.0.58
    Checking unicode-id v0.3.2
   Compiling crossbeam-utils v0.8.9
    Checking bitflags v1.3.2
   Compiling rayon-core v1.9.3
   Compiling proc-macro-hack v0.5.19
    Checking typed-arena v2.0.1
   Compiling typenum v1.[15](https://github.com/denoland/deno_ast/runs/6976588917?check_suite_focus=true#step:7:15).0
    Checking arrayvec v0.7.2
    Checking cpufeatures v0.2.2
    Checking fixedbitset v0.4.1
    Checking minimal-lexical v0.2.1
   Compiling semver v1.0.10
    Checking base64 v0.13.0
    Checking pathdiff v0.2.1
   Compiling swc_bundler v0.155.0
    Checking retain_mut v0.1.9
    Checking crc-catalog v1.1.1
    Checking relative-path v1.7.0
    Checking radix_fmt v1.0.0
    Checking bumpalo v3.10.0
    Checking diff v0.1.12
    Checking ansi_term v0.12.1
   Compiling lock_api v0.4.7
   Compiling indexmap v1.9.0
   Compiling num-traits v0.2.15
   Compiling num-integer v0.1.45
   Compiling num-bigint v0.4.3
   Compiling memoffset v0.6.5
   Compiling crossbeam-epoch v0.9.9
   Compiling rayon v1.5.3
   Compiling ahash v0.7.6
   Compiling generic-array v0.14.5
   Compiling proc-macro-error-attr v1.0.4
   Compiling proc-macro-error v1.0.4
    Checking tracing-core v0.1.27
   Compiling phf_shared v0.10.0
    Checking tinyvec v1.6.0
    Checking data-url v0.1.1
    Checking itertools v0.10.3
    Checking form_urlencoded v1.0.1
    Checking unreachable v0.1.1
    Checking better_scoped_tls v0.1.0
    Checking lexical-util v0.8.5
    Checking crc v2.1.0
    Checking pretty_assertions v1.2.1
    Checking unicode-normalization v0.1.19
    Checking debug_unreachable v0.1.1
    Checking lexical-parse-integer v0.8.6
    Checking lexical-write-integer v0.8.5
    Checking aho-corasick v0.7.18
    Checking nom v7.1.1
    Checking getrandom v0.2.7
    Checking num_cpus v1.13.1
    Checking time v0.1.44
    Checking idna v0.2.3
    Checking lexical-parse-float v0.8.5
    Checking lexical-write-float v0.8.5
    Checking crossbeam-channel v0.5.5
    Checking petgraph v0.6.2
    Checking ordered-float v2.10.0
    Checking regex v1.5.6
    Checking parking_lot v0.12.1
    Checking dashmap v5.3.4
   Compiling rand_core v0.6.3
    Checking lexical-core v0.8.5
    Checking block-buffer v0.10.2
    Checking crypto-common v0.1.3
    Checking crossbeam-deque v0.8.1
    Checking chrono v0.4.19
   Compiling pmutil v0.5.3
   Compiling darling_core v0.10.2
   Compiling Inflector v0.11.4
   Compiling tracing-attributes v0.1.21
   Compiling thiserror-impl v1.0.31
   Compiling swc_trace_macro v0.1.1
   Compiling rand_chacha v0.3.1
    Checking lexical v6.1.1
    Checking digest v0.10.3
   Compiling swc_macros_common v0.3.5
   Compiling swc_eq_ignore_macros v0.1.0
   Compiling static-map-macro v0.2.3
   Compiling auto_impl v0.5.0
   Compiling is-macro v0.2.1
   Compiling darling_macro v0.10.2
    Checking tracing v0.1.35
    Checking thiserror v1.0.31
   Compiling rand v0.8.5
    Checking sha-1 v0.10.0
   Compiling swc_visit_macros v0.3.1
   Compiling from_variant v0.1.3
   Compiling string_enum v0.3.1
   Compiling enum_kind v0.2.1
   Compiling swc_ecma_transforms_macros v0.5.0
   Compiling swc_config_macro v0.1.0
   Compiling swc_ecma_codegen_macros v0.7.1
    Checking st-map v0.1.6
    Checking swc_timer v0.6.0
   Compiling darling v0.10.2
    Checking string_cache v0.8.4
    Checking url v2.2.2
    Checking text_lines v0.4.1
   Compiling phf_generator v0.10.0
    Checking swc_visit v0.3.0
   Compiling ast_node v0.7.7
    Checking swc_config v0.1.1
    Checking sourcemap v6.0.2
   Compiling string_cache_codegen v0.5.2
   Compiling phf_macros v0.10.0
    Checking swc_common v0.18.9
   Compiling swc_atoms v0.2.12
   Compiling browserslist-rs v0.11.0
    Checking swc_fast_graph v0.6.0
    Checking swc_ecma_loader v0.30.2
    Checking phf v0.10.1
    Checking swc_graph_analyzer v0.7.0
    Checking swc_ecma_ast v0.79.0
    Checking swc_cached v0.1.1
    Checking preset_env_base v0.2.4
    Checking swc_ecma_visit v0.65.0
    Checking swc_ecma_parser v0.105.6
    Checking swc_ecma_codegen v0.109.1
    Checking dprint-swc-ext v0.2.0
    Checking swc_ecma_utils v0.86.2
    Checking swc_ecma_dep_graph v0.77.0
    Checking swc_ecma_transforms_base v0.89.5
error[E0277]: `R` cannot be shared between threads safely
   --> /home/runner/.cargo/registry/src/github.com-1ecc6299db9ec823/swc_ecma_transforms_base-0.89.5/src/rename/analyzer/scope.rs:191:18
    |
191 |                   .map(|child| {
    |  __________________^^^_-
    | |                  |
    | |                  `R` cannot be shared between threads safely
192 | |                     use std::collections::HashMap;
193 | |
194 | |                     let mut new_map = HashMap::default();
...   |
204 | |                     new_map
205 | |                 })
    | |_________________- within this `[closure@/home/runner/.cargo/registry/src/github.com-1ecc6299db9ec823/swc_ecma_transforms_base-0.89.5/src/rename/analyzer/scope.rs:191:22: 205:18]`
    |
    = note: required because it appears within the type `&R`
    = note: required because it appears within the type `[closure@/home/runner/.cargo/registry/src/github.com-1ecc6299db9ec823/swc_ecma_transforms_base-0.89.5/src/rename/analyzer/scope.rs:191:22: 205:18]`
note: required by a bound in `rayon::iter::ParallelIterator::map`
   --> /home/runner/.cargo/registry/src/github.com-1ecc6299db9ec823/rayon-1.5.3/src/iter/mod.rs:586:34
    |
586 |         F: Fn(Self::Item) -> R + Sync + Send,
    |                                  ^^^^ required by this bound in `rayon::iter::ParallelIterator::map`
help: consider further restricting this bound
    |
[16](https://github.com/denoland/deno_ast/runs/6976588917?check_suite_focus=true#step:7:16)7 |         R: Renamer + std::marker::Sync,
    |                    +++++++++++++++++++
For more information about this error, try `rustc --explain E0[27](https://github.com/denoland/deno_ast/runs/6976588917?check_suite_focus=true#step:7:28)7`.
error: could not compile `swc_ecma_transforms_base` due to previous error
Error: Process completed with exit code [101](https://github.com/denoland/deno_ast/runs/6976588917?check_suite_focus=true#step:7:102).
```

### Input code

_No response_

### Config

_No response_

### Playground link

_No response_

### Expected behavior

Should build.

### Actual behavior

_No response_

### Version

N/A

### Additional context

_No response_

--- a/crates/swc_ecma_transforms_base/src/rename/mod.rs
+++ b/crates/swc_ecma_transforms_base/src/rename/mod.rs
@@ -6,6 +6,10 @@ use swc_common::collections::AHashMap;
 use swc_ecma_ast::*;
 use swc_ecma_visit::{as_folder, noop_visit_mut_type, Fold, VisitMut, VisitMutWith, VisitWith};
 
+#[cfg(feature = "concurrent-renamer")]
+use self::renamer_concurrent::{Send, Sync};
+#[cfg(not(feature = "concurrent-renamer"))]
+use self::renamer_single::{Send, Sync};
 use self::{
     analyzer::Analyzer,
     collector::{collect_decls, CustomBindingCollector, IdCollector},
@@ -19,7 +23,7 @@ mod collector;
 mod eval;
 mod ops;
 
-pub trait Renamer: swc_common::sync::Send + swc_common::sync::Sync {
+pub trait Renamer: Send + Sync {
     /// Should reset `n` to 0 for each identifier?
     const RESET_N: bool;
 
@@ -242,4 +246,20 @@ where
 
         s.visit_mut_with(&mut rename_with_config(&map, self.config.clone()));
     }
+}
+
+#[cfg(feature = "concurrent-renamer")]
+mod renamer_concurrent {
+    pub use std::marker::{Send, Sync};
+}
+
+#[cfg(not(feature = "concurrent-renamer"))]
+mod renamer_single {
+    /// Dummy trait because swc_common is in single thread mode.
+    pub trait Send {}
+    /// Dummy trait because swc_common is in single thread mode.
+    pub trait Sync {}
+
+    impl<T> Send for T where T: ?Sized {}
+    impl<T> Sync for T where T: ?Sized {}
 }

